sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -233,6 +233,19 @@ public int hashCode() {
      */
     IterableAssert<T> inWindow(BoundedWindow window);
 
+    /**
+     * Creates a new {@link IterableAssert} like this one, but with the assertion restricted to only
+     * run on the provided window.
+     *
+     * <p>The assertion will expect outputs to be produced to the provided window exactly once. If
+     * the upstream {@link Trigger} may produce output multiple times, consider instead using {@link
+     * #inFinalPane(BoundedWindow)} or {@link #inOnTimePane(BoundedWindow)}.
+     *
+     * @return a new {@link IterableAssert} like this one but with the assertion only applied to the
+     *     specified window.
+     */
+    IterableAssert<T> inOnlyPane(BoundedWindow window);
```

Comment:
I double checked that adequate validation is in place to ensure this is not applied when there is more than one pane. Curiously the validation occurs on the `PaneInfo` of emitted elements rather than by examining the trigger. But anyhow this one seems safe enough.